frontend/packages/console-plugin-sdk/src/typings/pages.ts

```diff
@@ -24,7 +29,10 @@ namespace ExtensionProperties {
     mock?: boolean;
     /** The namespace scope. */
     namespace?: string;
-  }>;
+  }> & {
+    /** Function. If "true" is returned then particular item is included in the list. */
+    filter?: (filterGroups: ResourceListFilterGroups, obj: K8sResourceKind) => boolean;
```

Comment:
@mareklibra If I understand correctly, this PR is effectively about extending the `listFilters` object declared in `public/components/factory/list.tsx`, which is used to filter `List` component's rendered data.

The `ResourceListPage` extension makes Console aware of additional list pages, mapping model to page component to handle listing for that model. _The implementation of the entire list page component is provided by the plugin._

Therefore, I think a better approach is to make the `List` component itself extensible in terms of list filters used.

`ListInnerProps` already includes `filters`, this value is used to lookup the corresponding filter impl. within the `listFilters` object. So I'd suggest doing the following:

**Step 1, improve typing of list filters.** In the `List` component:
- add `ListFilterGroups` type (the `Resource` prefix is not needed)
- add type: `type ListFilter = (groups: ListFilterGroups, obj: K8sResourceKind) => boolean;`
- update `listFilters` object to use `ListFilter` type consistently, for example:
  - `'name': (filter, obj): ListFilter => fuzzyCaseInsensitive(filter, obj.metadata.name),`
- adapt `ListFilterGroups` type if necessary (add properties like `values` etc.)

**Step 2, allow extension of built-in list filters.** In the `List` component:
- in `ListInnerProps`, add `customFilterFuncs?: {[name: string]: ListFilter};`
- in `stateToProps`
  - add `customFilterFuncs = {}` to `props` object destructuring assignment
  - change `getFilteredRows(allFilters, data);` to `getFilteredRows(allFilters, data, customFilterFuncs);`
- in `getFilteredRows`
  - `const filter = listFilters[name] || customFilterFuncs[name];`
- in `filterPropType`
  - `if (key in listFilters || key in customFilterFuncs || key === 'loadTest') { continue; }`

**Step 3, use the newly introduced `customFilterFuncs` prop of `List` component in `VirtualMachinesPage`.**

This way, we have a `List` component that supports using custom filters in addition to ones that are built-in.

cc @christianvogt @spadgett

Reply:
Note that the `List` component is deprecated and soon to be removed! We should be making these changes to `table.tsx`.

Reply:
Agreed, I'll look into this.

What we actually need here is adding new prop to `Table` component (successor of `List`) that allows us to extend the built-in `tableFilters`.